### PR TITLE
Refactor PPO objective function

### PR DESCRIPTION
**Pull request by jon-tow**
This PR decouples the PPO loss function from the `AcceleratePPOModel` class for general reuse following @cat-state 's work in #75 . The following updates are also provided:

- Adds the unbiased KL-div estimates following http://joschu.net/blog/kl-approx.html
    - Related Issue: #53 
- Removes repeated initialization of `self.kl_ctl` in `accelerate_ppo_models` constructor.
- Removes `utils.modeling.clip_by_value` as this functionality was added to PyTorch nearly 2 years ago.
- Introduces more PPO statistics to the logger for greater debugging information.

Test Run: https://wandb.ai/jon-tow/ppo-test

### CODE REVIEW

**Comment by LouisCastricato** [comment]
This looks great on my side, @Dahoas im open to merging this tonight if this is ok for you too?

**Review by cat-state — APPROVED** [review]
LGTM!
